### DISCUSSION TITLE
Help! select bank with unused matrixadress button

### DISCUSSION THREAD

**Discussion by nendezkombet** [tttapa/Control-Surface#455] [Q&A] [answered]
Hi, Pieter !

i need some help, let's say i had 2x4 button matrix and i want use the left and right top corner as  increment decrement bank selector with led and the other button leave unused, so just two button used at the moment.

my question, it's possible replacing incrementdecrementbutton pin with matrix address for selecting bank ?

and it's look like i must write custom midi output element but i have no basic c++.
in the past few days i tried understanding the code writed in the controllsurface library (doxygen) and how to write/combine class for build my custom class but can't figure it out what i'm read.

So, can you explain some sample sketch ?

thx!

**Comment by tttapa** [comment, marked as answer]
I don't have much time right now, but you might want to have a look at this related issue: https://github.com/tttapa/Control-Surface/issues/308#issuecomment-723362162

Basically, if you want to be able to use the existing selectors or non-matrix MIDI elements, you need to implement the `ExtendedInputOutputElement` interface to tell Control Surface how it should read your button. The link above contains such an implementation.

Alternatively, you can read the matrix yourself in your sketch, you can then send the MIDI messages yourself, as shown in the Custom MIDI Output Element example and its Bankable counterpart, and you can change the banks manually (without a Selector) by calling [`bank.select(setting)`](You don't need to use a selector to change the bank, you can call bank.select(setting) directly, see https://tttapa.github.io/Control-Surface-doc/Doxygen/db/dbd/classBank.html#a801ab5dedea8200f1cdb26533cb6f78b). The implementation of the MIDI button matrix is really simple: https://github.com/tttapa/Control-Surface/blob/master/src/MIDI_Outputs/Abstract/MIDIButtonMatrix.hpp

Before diving into the Control Surface docs, it might be useful to familiarize yourself with the basics of C++ first, e.g. using https://learncpp.com.
